pkg/utils/parse_arf_result.go

```diff
@@ -724,17 +724,36 @@ func parseValues(remContent string, resultValues map[string]string) (string, []s
 	fixedText := remContent
 	if len(contentList) == 0 {
 		return remContent, valuesUsedList, valuesMissingList, nil
-	} // no processing needed, no urlencoded data
-	//process urlencoded content file data one by one and replace them in fixedText
+	}
+	// there are two types of content we need to process, one is url-encoded machine config source ex. {{ -a%20always0-F%20di }},
+	// the other one is not url_encoded ex. {{.var_some_value}}, that only needs value substition
+	// if the third charater start with `.` we know that the content is not url-encoded
+	// no processing needed, no {{ ... }} format found
+	//process {{ ... }} content file data one by one and replace them in fixedText
 	for _, content := range contentList {
 		//take out `{{ `,' }}' from content
-		trimmedContent := content[3:][:len(content)-6]
-		decodeContent, decodeErr := url.QueryUnescape(trimmedContent)
-		if decodeErr != nil {
-			return remContent, valuesUsedList, valuesMissingList, errors.Wrap(decodeErr, "error while decode remediation context: ")
+		//content starts with `{{.`` is not urlencoded
```

Comment:
What if... instead of checking via if statements, you always url-decode the string? As you already do with `QueryUnescape`. If the string stays the same or there was a decoding error, it means it wasn't urlencoded. If it succeeded unencoding it and the string is shorter than the original, it was urlencoded.

Reply:
That's a good way of doing it! that way we can take array of values as well